Platform/Plugins/com.tle.platform.common/src/com/tle/common/util/ExecUtils.java

```
@@ -92,12 +92,16 @@ public static void killLinuxProcessTree(int pid) {
         LOGGER.debug("getChildPid function did not run properly.\n" + errorOutput);
       }
       getChildPid.destroy();
-      // convert string to array of ints
-      pids =
-          Optional.of(
-              Arrays.stream(childPid.toString().replaceAll("\n", " ").split(" "))
-                  .mapToInt(Integer::parseInt)
-                  .toArray());
+
+      String childPidInfo = childPid.toString();
+      if (!childPidInfo.isEmpty()) {
```

Why would it be empty? Is it simply a normal situation, in that there were no child PIDs? Or it is always due to an error situation from going into the `else` block on line 89?

I'm just wondering if this needs a companion `else` block to either log the situation, or to throw an exception. :thinking: Or if it's just normal - i.e. there may be no child PIDs sometimes - then this is just a guard against this processing. Maybe a comment in this case would highlight the assumption. :thinking:

---

There may be no child PIDs sometimes - in fact most cases. PDF extraction causes ImageMagick to open Ghostscript which is the child PID - but for case in which ImageMagick can handle things fine on its own, such as most images, there will be no delegate process opened.

---

I feel we need to throw an runtime exception in the `else` block on line 89 if `getChildPid` fails for some unknown reasons.

---

Ah perhaps it's better to do a LOGGER.error rather than throwing an exception.

---

I agree on the LOGGER.warn.